### PR TITLE
feat(metrics): add relayer and monitor spend metrics

### PR DESCRIPTION
**Pull request by kevinhalliday**
Add total spend metrics to relayer and xfeemngr.

issue: #1962

### CODE REVIEW

File: monitor/xfeemngr/contract/bound.go [hunk 1]
```diff
@@ -128,14 +149,26 @@ func (c BoundFeeOracleV1) BulkSetFeeParams(ctx context.Context, params []binding
...
+	spendTotal.WithLabelValues(c.chain.Name, string(c.chain.NativeToken), methodBulkSetFeeParams).Add(totalSpentGwei(tx, rec))
+
 	return nil
 }
 
+// totalSpentGwei returns the total amount spent on a transaction in gwei.
+func totalSpentGwei(tx *ethtypes.Transaction, rec *ethtypes.Receipt) float64 {
+	fees := new(big.Int).Mul(rec.EffectiveGasPrice, new(big.Int).SetUint64(rec.GasUsed))
+	value := new(big.Int).Add(fees, tx.Value())
+	total := new(big.Int).Add(value, fees)
```
Comment (corverroos) [inline monitor/xfeemngr/contract/bound.go:166]: Not sure about the math here, feels like we are doing: `total := fees + tx.Value + fees`

File: monitor/xfeemngr/contract/bound.go [hunk 2]
```diff
@@ -128,14 +149,26 @@ func (c BoundFeeOracleV1) BulkSetFeeParams(ctx context.Context, params []binding
...
 	}
 
+	spendTotal.WithLabelValues(c.chain.Name, string(c.chain.NativeToken), methodBulkSetFeeParams).Add(totalSpentGwei(tx, rec))
+
 	return nil
 }
 
+// totalSpentGwei returns the total amount spent on a transaction in gwei.
+func totalSpentGwei(tx *ethtypes.Transaction, rec *ethtypes.Receipt) float64 {
+	fees := new(big.Int).Mul(rec.EffectiveGasPrice, new(big.Int).SetUint64(rec.GasUsed))
```
Comment (corverroos) [inline monitor/xfeemngr/contract/bound.go:164]: can use `umath.NewBigInt`

File: monitor/xfeemngr/contract/bound.go [hunk 3]
```diff
@@ -128,14 +149,26 @@ func (c BoundFeeOracleV1) BulkSetFeeParams(ctx context.Context, params []binding
...
+	value := new(big.Int).Add(fees, tx.Value())
+	total := new(big.Int).Add(value, fees)
+	totalGwei, _ := new(big.Int).Div(total, big.NewInt(int64(params.GWei))).Float64()
```
Comment (corverroos) [inline monitor/xfeemngr/contract/bound.go:167]: don't think the `int64()` cast is required actually.

**Review by corverroos — APPROVED** [review]

File: monitor/xfeemngr/contract/bound.go (hunk 1 above)
Comment (kevinhalliday) [reply monitor/xfeemngr/contract/bound.go:166]: yeah this is mistake

**Pull request merged**